File: C++/selection_sort.cpp
#include <iostream>
using namespace std;

#include <vector>

vector<int> selection_sort(vector<int> nums){
    /*
    Sorts an array using selection sort algorithm.
    Input:
    nums: Unsorted array
    Output:
    Sorted array 
    */
    
    int min_ind, temp;

    for (int i=0 ; i<nums.size() ; i++){
        min_ind = i;
        for (int j=i ; j<nums.size() ; j++){
            if (nums[j] < nums[min_ind]){
                min_ind = j;
            }
        }
        temp = nums[i];
        nums[i] = nums[min_ind];
        nums[min_ind] = temp;
    }
    return nums;
}

int main(){

    vector<int> nums = {10,5,2,3,8};
    vector<int> sorted_nums = selection_sort(nums);

    for (int i=0 ; i<sorted_nums.size() ; i++){
        cout << sorted_nums[i] << " ";
    }
    cout << endl;


    return 0;
}

File: Python/selection_sort.py
def selection_sort(nums):
    """
    Sorts an array using selection sort algorithm.
    Input:
    nums: Unsorted array
    Output:
    Sorted array 
    """

    for i in range(len(nums)):

        # find minimum of unsorted array
        min_ind = i
        for j in range(i,len(nums)):
            if nums[j] < nums[min_ind]:
                min_ind = j
        
        # perform swapping
        temp = nums[i]
        nums[i] = nums[min_ind]
        nums[min_ind] = temp
    
    return nums


if __name__ == "__main__":
    nums = [1,5,3,9,10,2]

    print(selection_sort(nums))